index-scheduler/src/batch.rs

```diff
@@ -431,27 +376,129 @@ impl IndexScheduler {
             Batch::Cancel(_) => todo!(),
             Batch::Snapshot(_) => todo!(),
             Batch::Dump(_) => todo!(),
-            Batch::DocumentClear { tasks, .. } => todo!(),
-            Batch::DocumentAddition {
+            Batch::IndexOperation(operation) => {
+                let index = match operation {
+                    IndexOperation::DocumentDeletion { ref index_uid, .. }
+                    | IndexOperation::DocumentClear { ref index_uid, .. } => {
+                        // only get the index, don't create it
+                        let rtxn = self.env.read_txn()?;
+                        self.index_mapper.index(&rtxn, &index_uid)?
+                    }
+                    IndexOperation::DocumentImport { ref index_uid, .. }
+                    | IndexOperation::Settings { ref index_uid, .. }
+                    | IndexOperation::DocumentClearAndSetting { ref index_uid, .. }
+                    | IndexOperation::SettingsAndDocumentImport { ref index_uid, .. } => {
+                        // create the index if it doesn't already exist
+                        let mut wtxn = self.env.write_txn()?;
+                        let index = self.index_mapper.index(&mut wtxn, index_uid)?;
+                        wtxn.commit()?;
+                        index
+                    }
```

Comment:
```suggestion
                        // TODO: check if the user was allowed to create an index.
                        
                        // create the index if it doesn't already exist
                        let mut wtxn = self.env.write_txn()?;
                        let index = self.index_mapper.index(&mut wtxn, index_uid)?;
                        wtxn.commit()?;
                        index
                    }
```

Sorry, that's something I forgot to forward in the `IndexOperation` / `Batch`, but we must keep it somewhere.
Also, maybe we should not batch tasks from a user that can create an index and a user that can't?
For now, I think this `TODO` is enough, but at some point, we'll need to look into that.